src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -95,7 +169,25 @@ protected internal override PinValue Read(int pinNumber)
 
         protected internal override void RemoveCallbackForPinValueChangedEvent(int pinNumber, PinChangeEventHandler callback)
         {
-            throw new NotImplementedException();
+            if (_pinNumberToEventHandler.TryGetValue(pinNumber, out LibGpiodDriverEventHandler eventHandler))
+            {
+                eventHandler.ValueFalling -= callback;
+                eventHandler.ValueRising -= callback;
+                if (eventHandler.IsCallbackListEmpty())
+                {
+                    if (_pinNumberToEventHandler.TryRemove(pinNumber, out eventHandler))
+                    {
+                        eventHandler.CancellationTokenSource.Cancel();
```

Comment:
this part of code should be locked - if someone adds a new event when we get here they will never get any events

Reply:
it is concurrent dictionary, do you think we still need a lock?

Reply:
concurrent dictionary only protects against not destroying the internals of the dictionary - not against the state itself. you might want to consider using regular dictionary and lock to have more control on when the locks happen